aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -261,18 +345,14 @@ impl<
     /// information (None if storage/pre-block version).
     /// If the layout of the resource is current UnSet, this function sets the layout of the
     /// group to the provided layout.
-    pub fn fetch_tagged_data(
+    pub fn convert_tagged_data(
         &self,
+        data_value: anyhow::Result<MVDataOutput<V>, MVDataError>,
         group_key: &K,
-        tag: &T,
-        txn_idx: TxnIndex,
     ) -> Result<(Version, ValueWithLayout<V>), MVGroupError> {
-        let key = (group_key.clone(), tag.clone());
-        let initialized = self.group_sizes.contains_key(group_key);
-
-        match self.values.fetch_data(&key, txn_idx) {
+        match data_value {
             Ok(MVDataOutput::Versioned(version, value)) => Ok((version, value)),
-            Err(MVDataError::Uninitialized) => Err(if initialized {
+            Err(MVDataError::Uninitialized) => Err(if self.group_sizes.contains_key(group_key) {
```

Comment:
initialized was being read before values read for a reason. Otherwise, we may have to repeat the read.